### DISCUSSION TITLE
Write the sql statement while generating sql migration

### DISCUSSION THREAD

**Discussion by FISHers6** [diesel-rs/diesel#3334] [Ideas]
Our company uses diesel to develop the database and extends the db tool through diesel cli. I want to do a function that can automatically generate migration. Can diesel-cli add a sql statement parameter when generating migration, so that the user can automatically complete the writing of migration. E.g `diesel migration generate migration_name --up up_sql --down down_sql`.

**Comment by weiznich** [comment]
It's not really clear for me what you imaging to be added to diesel. Maybe try to write some more concrete examples?

**Reply by FISHers6** [reply]
command like this:

`diesel migration generate create_posts --up 'CREATE TABLE IF NOT EXISTS posts {id BIGINT PRIMARY KEY}; --down 'DROP TABLE IF EXISTS posts'`

after execute this command, Diesel CLI create two files for us :

```
Creating migrations/20160815133237_create_posts/up.sql
Creating migrations/20160815133237_create_posts/down.sql
```

The content of the up.sql file is the parameters filled in by our command line
```
CREATE TABLE IF NOT EXISTS posts {
  id BIGINT PRIMARY KEY
};
```

The content of the `down.sql` file is

```
DROP TABLE IF EXISTS posts;
```

**Reply by weiznich** [reply]
Migrations usually contain longer sql statements. That makes it in my opinion a bad fit for command line arguments. Especially as they might contain several types of quotes which might be altered by your shell if not probably escaped. Therefore at least of my opinion adding such flags is not great for the usability of the command line tool, especially as the described functionality could be easily emulated by just piping the corresponding commands to the right files after creating the migration.

**Reply by FISHers6** [reply]
I agree with your point of view, I also think so, I think when using these two new parameters, the scene may not be manually calling the diesel command line tool directly. In most scenarios, the diesel cli is called by implementing a db tools program, and the programmer needs to ensure the content format specification of the written file. Of course, we can also write content to the file by other means.thank you for your reply~
